pkg/core/tokens/revocations.go

```diff
@@ -0,0 +1,66 @@
+package tokens
+
+import (
+	"context"
+	"strings"
+
+	system_proto "github.com/kumahq/kuma/api/system/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/system"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
+)
+
+// Revocations keeps track of revoked tokens.
+// If only one token is compromised, it's more convenient to revoke it instead of rotate signing key and regenerate all tokens.
+// Revocation list is stored as Secret (in case of mesh scoped tokens) or GlobalSecret (global scoped tokens).
+// IDs of token are stored in secret in comma separated format: "id1,id2".
+type Revocations interface {
+	IsRevoked(id string) (bool, error)
+}
+
+func NewRevocations(manager manager.ReadOnlyResourceManager, revocationKey core_model.ResourceKey) Revocations {
+	return &secretRevocations{
+		manager:       manager,
+		revocationKey: revocationKey,
+	}
+}
+
+type secretRevocations struct {
+	manager       manager.ReadOnlyResourceManager
+	revocationKey core_model.ResourceKey
+}
+
+func (s *secretRevocations) IsRevoked(id string) (bool, error) {
+	data, err := s.getSecretData()
+	if err != nil {
+		return false, err
+	}
+	if len(data) == 0 {
+		return false, nil
+	}
+	rawIds := strings.TrimSuffix(string(data), "\n")
+	ids := strings.Split(rawIds, ",")
```

Comment:
It may be more robust, but it would be less convenient for users.
If we were to change format to JSON, then we would need to add some kumactl tooling, to construct secrets for users.
So far we were trying to avoid involving kumactl when it comes to operating kuma on Kubernetes.

Reply:
I'm taking this as optional since you approved a PR. If you feel strongly about this change, please let me know so we can work on that.